src/Core/Domain/Customer/Group/ValueObject/GroupName.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Core\Domain\Customer\Group\ValueObject;
+
+class GroupName
```

Comment:
This is useless IMO, it can be just a string 🤷‍♂️

Reply:
unless you validate length and regex, hen this VO won't be useless anymore (as I see in Group.php there are rules
`'validate' => 'isGenericName', 'required' => true, 'size' => 32`).

Reply:
Yes, but now there's no validation at all.

Reply:
| Yes, but now there's no validation at all.

yes, thats why it should be added :smile: and then the VO becomes usefull and does what it is designed to do.

btw when validation is added, there should appear a constants in this VO- VALID_REGEX_PATTERN and MAX_LENGTH, which can later be reused in customerGroup from type when adding form validation (and alos the MAX_LENGTH constant value can be reused in Group.php instead of now hardcoded number 32).

Reply:
We shouldn't use some VOs for any field, the initial purpose of a VO is to identify an entity like `ProductId` `GroupId` so it can include some checking to make sure the provided ID format is at least legit before trying to use it

We also extended the usage of VO to other fields that identify some configurations, although strictly speaking it's already out of the scope of a VO, like `PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductType` which fires checking the value of an enum. This kind of VO could also be accepted.

But we shouldn't use VOs everywhere and especially not on fully dynamic properties like names. This is pushing the complexity too far, the validation of this field must be handled by a Validator service dedicated to the Group entity, not via a multitude of VO objects.

Reply:
TLDR; VOs are useful in some use cases but they shouldn't replace the validation every time, so no need to use them everywhere

Reply:
it is recommended to use VO's in such cases (if you google it, there are sayings like "you shouldn't be afraid of object and should get rid as many native type as possible. That is OOP.), but maybe 2 simple validation rules are not enough to make it worth using it here, so I agree it is probably impractical (at least for this project?) :man_shrugging: 

Anyway when this class is deleted we can use some CustomerGroupSettings class to contain those constants for validation (as we did in several other places)